File: leetcode_2022/16.levelOrderBinTree.py
def levelOrder(root):
    if not root:
        return []

    result = []
    queue = [root]
    
    while len(queue):
        length = len(queue)
        count = 0
        currentLevelValues = []
        
        while count < length:
            currentNode = queue.pop(0)
            currentLevelValues.append(currentNode.val)
            
            if currentNode.left:
                queue.append(currentNode.left)    
                
            if currentNode.right:
                queue.append(currentNode.right)    
                
            count += 1
        
        result.append(currentLevelValues)

    return result

File: leetcode_2022/udemy_leetcode/16.levelOrderBinTree.js
// given a binary tree, return the level order traversal of the nodes' values as an array.
// use BFS... 


//identify level in tree
//initialize subarray
//push finished subarray into result

const levelOrder = (root) => {
    if(!root) return []
    const result = []
    const queue = [root]

    while(queue.length){
        let length = queue.length
        let count = 0
        const currentLevelValues = []

        while (count < length){
            const currentNode = queue.shift()
            currentLevelValues.push(currentNode.val)

            if(currentNode.left){
                queue.push(currentNode.left)
            }

            if(currentNode.right){
                queue.push(currentNode.right)
            }
            count++
        }
        result.push(currentLevelValues)
    }
    return result
}
